### PR TITLE
Operation refactor to support diverse operation types

### PR DESCRIPTION
**Pull request by kfangw**
### Github Issue
#311 #439 

### Background
* BlockOperation Hash means TxHash + OpHash.
* Needs diverse Operations

### Solution
* Refactor: Use json.Unmarshaler interface
* BlockOperation store `body` instead of `amount, target`. Because these fields are not mandatory for all operations. Some operation has no amount or target
* Introduce OpHash for BlockOperation
* Introduce `OperationBodyPayable` interface for `OperationBodyCreateAccount` and `OperationBodyPayment`
* Add dummy OperationBody for congress voting

### CODE REVIEW

**Comment by kfangw** [comment]
please review Commit by Commit.
@spikeekips @Geod24 @Charleslee522 

I will make another PR for Congress voting on top of this PR.
There might be a lot of concerns for Congress voting operations such as validation.

**Comment by kfangw** [comment]
> Everything looks good 👍 One thing I suggest that instead of Body []byte in BlockOperation, how about using Body transaction.OperationBody. It will be automatically stored as serialized into storage and also without unmarshaling can be used.

I am planning to refactor packages like this
The `block` package does not depend on any other packages except `common` and `storage`.
IMO, the `block` package is a model(or schema) of database.
Hence, it should not import other packages. But other packages use it as a data entity.
If it is accepted, packages are more `loose coupling high cohesion`

But now, It is convenience if the BlockOperation knows `transaction.OperationBody`.
So I will patch it according to your comment
The refactoring will be done after several weeks later.

@spikeekips what do you think about this?

**Comment by spikeekips** [comment]
@kfangw 👍

**Comment by Charleslee522** [comment]
@kfangw Okay. I'll check test case in next pr. Please check integration test fail.

File: lib/node/runner/api_block.go [hunk 1]
```diff
@@ -179,7 +179,9 @@ func UnmarshalNodeItemResponse(d []byte) (itemType NodeItemDataType, b interface
 		err = unmarshal(&t)
 		b = t
 	case NodeItemTransaction:
-		b, err = transaction.NewTransactionFromJSON(d[len(sc.Bytes())+1:])
+		var t transaction.Transaction
+		err = unmarshal(&t)
+		b = t
```
Comment (Geod24) [inline lib/node/runner/api_block.go]: That looks like a mistake

Comment (kfangw) [reply lib/node/runner/api_block.go]: It is intended.

Comment (Geod24) [reply lib/node/runner/api_block.go]: Oh I see what you did! Yeah that makes sense then! 👍

**Pull request merged**